reconcile/dashdotdb_dora.py

```diff
@@ -0,0 +1,480 @@
+
+                if not result:
+                    break
+
+                records.extend(result)
+
+                offset += limit
+
+            if not records:
+                LOG.info("No deployments found for %s", app_env)
+
+            return [Deployment(row[0], row[1]) for row in records]
+
+
+@dataclass
+class Commit:
+    repo: str
+    sha: str
+    date: datetime
+
+    def lttc(self, finish_timestamp: datetime) -> int:
+        commit_date_tzaware = self.date
+        finish_timestamp_tzaware = finish_timestamp
+
+        if commit_date_tzaware.tzinfo is None:
+            commit_date_tzaware = commit_date_tzaware.replace(tzinfo=timezone.utc)
+
+        if finish_timestamp_tzaware.tzinfo is None:
+            finish_timestamp_tzaware = finish_timestamp_tzaware.replace(
+                tzinfo=timezone.utc
+            )
+
+        return int((finish_timestamp_tzaware - commit_date_tzaware).total_seconds())
+
+
+@dataclass(eq=True, frozen=True)
+class SaasTarget:
+    app_name: str
+    env_name: str
+    path: str
+    resource_template: str
+    namespace: str
+    pipeline: str
+
+    @property
+    def app_env(self) -> AppEnv:
+        return AppEnv(self.app_name, self.env_name)
+
+
+@dataclass(eq=True, frozen=True)
+class RepoChanges:
+    repo_url: Optional[str]
+    ref_from: Optional[str]
+    ref_to: Optional[str]
+
+
+@dataclass(eq=True, frozen=True)
+class SummaryKey:
+    rt_name: str
+    target_ns: str
+
+
+@dataclass(eq=True, frozen=True)
+class SummaryEntry:
+    repo_url: str
+    target_ref: str
+
+
+class DashdotdbDORA(DashdotdbBase):
+    def __init__(self, dry_run: bool, thread_pool_size: int = 5) -> None:
+        settings = queries.get_app_interface_settings()
+        vault_settings = get_app_interface_vault_settings()
+        secret_reader = create_secret_reader(use_vault=vault_settings.vault)
+
+        # init dashdotdb
+        super().__init__(
+            dry_run=dry_run,
+            thread_pool_size=thread_pool_size,
+            marker="DDDB_DORA:",
+            scope="dora",
+            secret_reader=secret_reader,
+        )
+
+        # init dep db
+        depdb_secret = secret_reader.read_all_secret(DEPLOYMENT_DB_SECRET)
+        self.dep_db = DeploymentDB(
+            depdb_secret["db.host"],
+            depdb_secret["db.port"],
+            depdb_secret["db.name"],
+            depdb_secret["db.user"],
+            depdb_secret["db.password"],
+        )
+
+        # init github secret
+        gh_instance = queries.get_github_instance()
+        self.gh_token = secret_reader.read(gh_instance["token"])
+
+        # init app_interface_get_file
+        gl_instance = queries.get_gitlab_instance()
+        gl = GitLabApi(
+            gl_instance, project_id=13582, settings=settings
+        )  # 13582 is app-interface
+
+        self.app_interface_get_file = functools.cache(gl.get_file)
+
+        # this is the RAW GitLab API, not our wrapper GitLabApi
+        self.gl = gl.gl
```

Comment:
I think this is also not applicable for integrations like this one that run on a cron.

Reply:
Similar concern to https://github.com/app-sre/qontract-reconcile/pull/3793#discussion_r1329876129, it's an easy fix to avoid leak then we don't need to worry about change it from cronjob to a deployment later.

Reply:
Fixed